woocommerce/wc-admin
# Adding Navigation Favorites REST API Endpoints

Fixes #6136 
Adding simple CRUD API to manage Navigation Favorites:

| Endpoint | Params | Method |
| --- | ----------- | ----------- |
| `navigation/favorites` (read all) | `user_id` | `GET`
| `navigation/favorites` (create) | `item_id`, `user_id` | `POST`
| `navigation/features` (delete) | `item_id`, `user_id` | `DELETE`

All endpoints accept the `user_id` optional parameter to specify user for favorites operations, and it will default to the current user when this is not provided. 

### Questions
- Are permissions handled correctly here? Trying to pattern after a few others I saw in the APIs, but not entirely sure I understand all the conventions, or what is appropriate for this use case.
- Experimenting with a little exception handling, although I haven't done this in PHP before, so let me know if anything looks amiss (or if we'd rather not throw exceptions at all).

### To-do
- Adding tests (doing so now)

_Note:_ This is currently based off of the unmerged `add/6131-api-favorites-registration` branch.

### Detailed test instructions:

-   Checkout branch
-   Use _Postman_ or other tools to send requests to `/wp-json/wc-admin/navigation/favorites`, using the `GET`, `POST`, and `DELETE` methods. Ensure the request is authenticated with the nonce and cookie headers.
- Trying adding an item, listing them, and deleting. 
- Ensure no unhandled errors or unexpected responses.
- Then add the `user_id` parameter, and run through the specified methods again. 
- Follow along with the `woocommerce_admin_navigation_favorites` in the `wp_usermeta` table to ensure the correct user/etc is being altered.
